js/ui/splitter.js

```diff
@@ -0,0 +1,163 @@
+import $ from "../core/renderer";
+import Widget from "./widget/ui.widget";
+import domAdapter from "../core/dom_adapter";
+import eventsEngine from "../events/core/events_engine";
+import pointerEvents from "../events/pointer";
+import { addNamespace } from "../events/utils";
+import { getWindow } from "../core/utils/window";
+import { isString } from "../core/utils/type";
+
+const SPLITTER_CLASS = "dx-splitter";
+const SPLITTER_WRAPPER_CLASS = `${SPLITTER_CLASS}-wrapper`;
+const SPLITTER_TRANSPARENT_CLASS = `${SPLITTER_CLASS}-transparent`;
+const SPLITTER_BORDER_CLASS = `${SPLITTER_CLASS}-border`;
+
+const SPLITTER_MODULE_NAMESPACE = "dxSplitterResizing";
+const SPLITTER_POINTER_DOWN_EVENT_NAME = addNamespace(pointerEvents.down, SPLITTER_MODULE_NAMESPACE);
+const SPLITTER_POINTER_MOVE_EVENT_NAME = addNamespace(pointerEvents.move, SPLITTER_MODULE_NAMESPACE);
+const SPLITTER_POINTER_UP_EVENT_NAME = addNamespace(pointerEvents.up, SPLITTER_MODULE_NAMESPACE);
+const SPLITTER_WINDOW_RESIZE_EVENT_NAME = addNamespace("resize", SPLITTER_MODULE_NAMESPACE);
+
+export default class SplitterControl extends Widget {
+    _initMarkup() {
+        this._$container = this.option("container");
+        this._$leftElement = this.option("leftElement");
+        this._$rightElement = this.option("rightElement");
+        this._onApplyPanelSize = this._createActionByOption("onApplyPanelSize");
+
+        this.$element()
+            .addClass(SPLITTER_WRAPPER_CLASS);
+        this._$splitterBorder = $("<div>")
+            .addClass(SPLITTER_BORDER_CLASS)
+            .appendTo(this.$element());
+        this._$splitter = $("<div>")
+            .addClass(SPLITTER_CLASS)
+            .addClass(SPLITTER_TRANSPARENT_CLASS)
+            .appendTo(this._$splitterBorder);
+    }
+
+    _render() {
+        this._detachEventHandlers();
+        this._attachEventHandlers();
+    }
+
+    _clean() {
+        this._detachEventHandlers();
+        super._clean();
+    }
+
+    _attachEventHandlers() {
+        const document = domAdapter.getDocument();
+        eventsEngine.on(this._$splitter, SPLITTER_POINTER_DOWN_EVENT_NAME, this._onMouseDownHandler.bind(this));
+        eventsEngine.on(document, SPLITTER_POINTER_MOVE_EVENT_NAME, this._onMouseMoveHandler.bind(this));
+        eventsEngine.on(document, SPLITTER_POINTER_UP_EVENT_NAME, this._onMouseUpHandler.bind(this));
+        eventsEngine.on(getWindow(), SPLITTER_WINDOW_RESIZE_EVENT_NAME, this._windowResizeHandler.bind(this));
```

Comment:
Widget have already a subscription to the window.resize event. And it has a `_dimensionChanged` as a handler for this event. I think we should use this method instead of the new one.